### PR TITLE
Add custom attribute support to InvocationContext

### PR DESCRIPTION
**Pull request by isurulucky**

### CODE REVIEW

File: tests/ballerina-test/src/test/resources/test-src/runtime/invocation-context.bal [hunk 1]
```diff
@@ -43,3 +43,14 @@ function testAuthToken () returns (boolean) {
     runtime:getInvocationContext().authContext.authToken = authToken;
     return authToken == runtime:getInvocationContext().authContext.authToken;
 }
+
+function testAttributes () returns (boolean) {
```
Comment (riyafa) [inline tests/ballerina-test/src/test/resources/test-src/runtime/invocation-context.bal]: Brackets around boolean can be removed

File: tests/ballerina-test-integration/src/test/java/org/ballerinalang/test/invocationcontext/InvocationContextTestCase.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+import org.ballerinalang.test.util.HttpResponse;
+import org.ballerinalang.test.util.TestConstant;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+import java.io.File;
+import java.util.HashMap;
+import java.util.Map;
+
+public class InvocationContextTestCase extends IntegrationTestCase {
```
Comment (riyafa) [inline tests/ballerina-test-integration/src/test/java/org/ballerinalang/test/invocationcontext/InvocationContextTestCase.java]: Shall we add class documentation?

File: tests/ballerina-test-integration/src/test/resources/invocationcontext/invocation-context-test.bal [hunk 3]
```diff
@@ -0,0 +1,47 @@
+import ballerina/http;
+import ballerina/log;
+import ballerina/runtime;
+import ballerina/io;
```
Comment (anupama-pathirage) [inline tests/ballerina-test-integration/src/test/resources/invocationcontext/invocation-context-test.bal]: Do we need io package here

File: tests/ballerina-test/src/test/resources/test-src/runtime/invocation-context.bal (hunk 1 above)
Comment (isurulucky) [reply tests/ballerina-test/src/test/resources/test-src/runtime/invocation-context.bal]: Fixed with https://github.com/ballerina-platform/ballerina-lang/pull/9069.

File: tests/ballerina-test-integration/src/test/java/org/ballerinalang/test/invocationcontext/InvocationContextTestCase.java (hunk 2 above)
Comment (isurulucky) [reply tests/ballerina-test-integration/src/test/java/org/ballerinalang/test/invocationcontext/InvocationContextTestCase.java]: Fixed in https://github.com/ballerina-platform/ballerina-lang/pull/9069.

File: tests/ballerina-test-integration/src/test/resources/invocationcontext/invocation-context-test.bal (hunk 3 above)
Comment (isurulucky) [reply tests/ballerina-test-integration/src/test/resources/invocationcontext/invocation-context-test.bal]: No we don't, fixed with https://github.com/ballerina-platform/ballerina-lang/pull/9071.

**Pull request merged**